### PR TITLE
implement IsolatedEvent command, add IsolatedEvent

### PR DESCRIPTION
**Pull request by gmengqi**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -221,7 +221,7 @@ Examples:
 Creates a weekly recurring event or a non-recurring event
 
 #### 1) Non-recurring Event
-Format: `event create e/EVENT_NAME f/START_DATE t/END_DATE`
+Format: `event create INDEX en/EVENT_NAME f/START_DATE t/END_DATE`
```
Comment (clydelhui) [inline docs/UserGuide.md:224]: Please edit the command format! 👍

File: src/main/java/seedu/address/logic/commands/AddIsolatedEventCommand.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    public static final String MESSAGE_SUCCESS = "New isolated event added: %1$s";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Add an isolated event into the isolated event list"
+            + "by the index number used in the last person listing. "
+            + "Existing remark will be overwritten by the input.\n"
+            + "Parameters: start date and end date (must be in the format of ) "
+            + "e/ [REMARK]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "r/ Likes to swim.";
```
Comment (clydelhui) [inline src/main/java/seedu/address/logic/commands/AddIsolatedEventCommand.java:29]: This usage text seems to be from `Remark`. Perhaps you could update it to the command syntax for isolated events!

File: src/main/java/seedu/address/logic/parser/IsolatedEventCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * Parser class for Isolated Event
+ */
+public class IsolatedEventCommandParser implements Parser<AddIsolatedEventCommand> {
+
+    /**
+     * Parse the IsolatedEvent command.
+     * @param args
+     * @return AddIsolatedEventCommand
+     * @throws ParseException
```
Comment (clydelhui) [inline src/main/java/seedu/address/logic/parser/IsolatedEventCommandParser.java:27]: Perhaps the Javadocs could include some description on what the parameters, return values and exceptions mean?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -110,6 +114,41 @@ public static Tag parseTag(String tag) throws ParseException {
         return new Tag(trimmedTag);
     }
 
+    /**
+     * Parses a {@code String name} into a {@code String name} into the correct format.
+     * @param eventName
+     * @return a String object
+     * @throws ParseException
+     */
```
Comment (clydelhui) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:122]: Perhaps this Javadoc could also explain the parameters, return values and exceptions

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 5]
```diff
@@ -110,6 +114,41 @@ public static Tag parseTag(String tag) throws ParseException {
...
+        }
+        return trimmedEventName;
+    }
+
+    /**
+     * Parses a {@code String date} into a {@code LocalDateTime date} into the correct format.
+     * @param date
+     * @return String object
+     * @throws ParseException
+     */
```
Comment (clydelhui) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:138]: Perhaps this Javadoc could also explain the parameters, return values and exceptions

File: src/main/java/seedu/address/model/event/IsolatedEvent.java [hunk 6]
```diff
@@ -1,15 +1,48 @@
...
+    private final String eventName;
+    private final LocalDateTime startDate;
+    private final LocalDateTime endDate;
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+
+    /**
+     * Constructor for IsolatedEvent class.
+     * @param eventName
+     * @param startDate
+     * @param endDate
```
Comment (clydelhui) [inline src/main/java/seedu/address/model/event/IsolatedEvent.java:26]: Perhaps this Javadoc could also explain the parameters, return values and exceptions

File: src/main/java/seedu/address/model/event/IsolatedEvent.java [hunk 7]
```diff
@@ -28,6 +61,23 @@ public int compareTo(IsolatedEvent o) { //TODO: Add implementation for compareTo
     @Override
     public boolean occursBetween(LocalDateTime startPeriod, LocalDateTime endPeriod) {
         // TODO: Implement this method.
```
Comment (clydelhui) [inline src/main/java/seedu/address/model/event/IsolatedEvent.java:63]: Perhaps the `TODO` comments can be deleted 👍

File: src/main/java/seedu/address/model/person/Person.java [hunk 8]
```diff
@@ -52,6 +55,14 @@ public Address getAddress() {
         return address;
     }
 
+    public void addIsolatedEvent(IsolatedEvent event) {
+        isolatedEventList.insert(event);
+    }
+
+    public IsolatedEventList getIsolatedEventList() {
+        return isolatedEventList;
+    }
```
Comment (clydelhui) [inline src/main/java/seedu/address/model/person/Person.java:64]: I like the use of abstraction here! 💯

File: src/test/java/seedu/address/model/event/IsolatedEventListTest.java [hunk 9]
```diff
@@ -32,10 +28,31 @@ void insert_nullEvent_throwsNullPointerException() {
...
     void testToString() {
-        isolatedEventList.insert(new IsolatedEventStub("Biking"));
-        isolatedEventList.insert(new IsolatedEventStub("Skiing"));
-        isolatedEventList.insert(new IsolatedEventStub("Canoeing"));
+        isolatedEventList.insert(new IsolatedEventStub("Biking", "09/03/2023 14:00",
+                "09/03/2023 15:00"));
+        isolatedEventList.insert(new IsolatedEventStub("Skiing", "09/03/2023 14:00",
+                "09/03/2023 15:00"));
+        isolatedEventList.insert(new IsolatedEventStub("Canoeing", "09/03/2023 14:00",
+                "09/03/2023 15:00"));
```
Comment (clydelhui) [inline src/test/java/seedu/address/model/event/IsolatedEventListTest.java:36]: Using utils here might be good!

**Review by clydelhui — COMMENTED** [review]
Overall good code quality! Some minor documentation issues and organization suggestions!

File: src/main/java/seedu/address/model/event/IsolatedEvent.java [hunk 10]
```diff
@@ -27,7 +45,28 @@ public int compareTo(IsolatedEvent o) { //TODO: Add implementation for compareTo
...
+        }
+
+        if (endDate.isAfter(endPeriod)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static boolean isValidEventName(String eventName) {
```
Comment (Alvozyn) [inline src/main/java/seedu/address/model/event/IsolatedEvent.java:59]: I think we can put this line into the event class

**Pull request merged**